lessons/cryptography/cryptography-hashing-checksums-salts.md

```diff
@@ -0,0 +1,82 @@
+# [Cryptography] Hashing, Checksums, and Salts
+*What is data integrity and how is it verified?*
+
+![banner alt text](.rsrc/banner.png)
+
+# Objectives
+- Student can describe why checksums are used: data integrity, errors in data transmission can occur
+- Student can describe why hashes are used: data integrity, improving security of storage of passwords
+- Student can distinguish between hashes and checksums
+- Student can identify key the following properties of cryptographic hashing functions: non-reversibility (hashing should be a one way function, meaning you can't get the original input from the hash), diffusion or avalanche effect (making a small change in the input should result in a large change in the hash), determinism (the same string should generate the same hash), uniformity or collision avoidance (should be a good distribution), non-predictability (you should not be able to guess the hash from the string)
+- Student recognizes some of the popular hashing algorithms: MD5, SHA family
+- Student can define what a salt is and why it is important in the context of password storage (salts are a randomized chuck of data that is added to what you are hashing --> this will cause 2 passwords that are the same to have 2 different hashes)
+
+# Introduction
+Introduction between 100-150 words.
+
+1. Cryptographic Hashes
+2. Common Cryptographic Hashes
+3. Applications of Cryptographic Hashes
+4. Password Salting
```

Comment:
Salting & Peppering 

Talk about the difference between the two. Also some options for graphics and such here I think.